File: Java/1456. Maximum Number of Vowels in a Substring of Given Length.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int maxVowels(String s, int k) {
        Set<Character> sesliHarfler = new HashSet<>();
        for (char c : new char[] { 'a', 'e', 'i', 'o', 'u' }) {
            sesliHarfler.add(c);
        }
        
        int bulunanMaxSesliHarf = 0;
        int sayac = 0;

        for (int i = 0; i < s.length(); i++) {
            if (sesliHarfler.contains(s.charAt(i)))
                sayac++;

            if (i >= k && sesliHarfler.contains(s.charAt(i - k)))
                sayac--;

            if (i >= k - 1)
                bulunanMaxSesliHarf = Math.max(bulunanMaxSesliHarf, sayac);
        }

        return bulunanMaxSesliHarf;
    }
}

File: CSharp/1456. Maximum Number of Vowels in a Substring of Given Length.cs
public class Solution
{
    public int MaxVowels(string s, int k)
    {
        HashSet<char> sesliHarfler = new HashSet<char> { 'a', 'e', 'i', 'o', 'u' };
        int bulunanMaxSesliHarf = 0, sayac = 0;

        //karakterleri sırayla gez
        for (int i = 0; i < s.Length; i++)
        {
            // sesli ise sayacı artır
            if (sesliHarfler.Contains(s[i]))
                sayac++;

            // sol tarafı ilerlet (k’yı geçtiyse)
            if (i >= k && sesliHarfler.Contains(s[i - k]))
                sayac--;

            // i >= k’ya ulaştıysa bulunanMaxSesliHarf güncelle
            if (i >= k - 1)
                bulunanMaxSesliHarf = Math.Max(sayac, bulunanMaxSesliHarf);
        }

        return bulunanMaxSesliHarf;
    }
}